server/src/main/java/io/druid/client/cache/MapCache.java

```diff
@@ -103,7 +103,10 @@ public void put(NamedKey key, byte[] value)
   {
     Map<NamedKey, byte[]> retVal = Maps.newHashMap();
     for (NamedKey key : keys) {
-      retVal.put(key, get(key));
+      final byte[] value = get(key);
+      if (value != null) {
+        retVal.put(key, value);
+      }
```

Comment:
We should add it to javaDoc of getBulk that its expected to return a map with no null values.